colossus-metrics/src/main/scala/colossus/metrics/Collection.scala

```diff
@@ -142,169 +23,89 @@ trait TickedCollector extends EventCollector {
-   */
-  def local(params: P, collectorConfig: CollectorConfig): C with LocalLocality
-
-  /**
-   * Generate a shared collector given the params object
-   */
-  def shared(params:P, collectorConfig: CollectorConfig)(implicit collector: ActorRef): C with SharedLocality
-}
-
-
-
-trait Collection[L <: Locality] {
-  import Collection.ParamsFor
-  //notice - the type lambda is basically the same as a view bounds (P <%
-  //MetricParams[U]), but view bounds are deprecated.  We need this instead of
-  //just subtype (P <: MetricParams[U]) becuase for some reason type inference
-  //fails when we do that
-  def getOrAdd[T <: EventCollector : ClassTag, P : ParamsFor[T]#Type](params: P)(implicit generator: Generator[T,P]): T with L
+trait Collector {
 
+  def address: MetricAddress
+  def tick(interval: FiniteDuration): MetricMap
 }
-object Collection {
-  type ParamsFor[U <: EventCollector] = ({type Type[P] = P => MetricParams[U,P]})
-}
-
-
-class DuplicateMetricException(message: String) extends Exception(message)
-
-/**
- * A Local collection is designed only to be used inside an actor or otherwise
- * thread-safe environment.  Notice that even though the collection we're
- * storing the event collectors is thread-safe, the collectors themselves are
- * not.  The collector actor must be the actor receiving events for metrics in
- * the map
- *
- * note - this is not intended to be directly constructed by users
- */
-class LocalCollection(
-  namespace: MetricAddress = MetricAddress.Root,
-  val globalTags: TagMap = TagMap.Empty, 
-  val intervals: Seq[FiniteDuration] = Nil,
-  metrics:ConcurrentHashMap[MetricAddress, Local[EventCollector]] = new ConcurrentHashMap[MetricAddress, Local[EventCollector]],
-  parent: Option[LocalCollection] = None
-) extends Collection[LocalLocality] {
-  import Collection.ParamsFor
-  import LocalCollection._
 
-  val localProducers = collection.mutable.ArrayBuffer[MetricProducer]()
+class CollectionMap[T] {
 
-  val collectorConfig = CollectorConfig(intervals)
+  private val map = new ConcurrentHashMap[T, AtomicLong]
 
-
-  def getOrAdd[T <: EventCollector : ClassTag, P : ParamsFor[T]#Type](_params: P)(implicit creator: Generator[T, P]): Local[T] = {
-    val params: P = _params.transformAddress(namespace / _)
-    parent.map{_.getOrAdd(params)}.getOrElse {
-      val created = creator.local(params, collectorConfig)
-      metrics.putIfAbsent(params.address, created: Local[EventCollector]) match {
-        case null => created
-        case exists: T => {
-          //because of type erasure, if we try to match against
-          //Local[T] or T with LocalLocality[T], it creates false positives (it
-          //thinks a Local[Counter] is the same type as a Local[Rate]), bad :( 
-          exists.asInstanceOf[Local[T]] 
-        }
-        case other => throw new DuplicateMetricException(s"An event collector of type ${other.getClass.getSimpleName} already exists")
+  def increment(tags: T, num: Long = 1) {
```

Comment:
small comment: this function is nearly identical to the one underneath it, maybe merge the two?